src/_pytest/mark/__init__.py

```diff
@@ -224,6 +225,7 @@ def deselect_by_mark(items, config):
         return
 
     try:
+        matchexpr = " and ".join(expr.lstrip() for expr in matchexpr)
```

Comment:
This is problematic BTW, e.g.

```
-m 'a or b' -m 'c'
```

would become

```
a or b and c    ===     a or (b and c)
```

when it should be

```
(a or b) and c
```

The better way would be to compile & evaluate the expressions separately and combine the results.